i18n4k-generator/src/main/kotlin/de/comahe/i18n4k/generator/I18n4kGenerator.kt

```diff
@@ -179,44 +153,132 @@ class I18n4kGenerator(
             }
         }
 
+        fun addFactoryClass(className: ClassName, key: String, index: Int, params: Map<String, String?>) {
```

Comment:
Can you add as AsciiDoc file in [i18n4k/doc](https://github.com/comahe-de/i18n4k/tree/master/doc) where you describe your custom factories and the type inside the parameter name?